pkg/api/types.go

```diff
@@ -905,7 +905,7 @@ type ReplicationControllerSpec struct {
 	Replicas int `json:"replicas"`
 
 	// Selector is a label query over pods that should match the Replicas count.
-	Selector map[string]string `json:"selector"`
+	Selector labels.Selector `json:"selector"`
```

Comment:
labels.Selector is an interface? The underlying representation is the string, or LabelSelector (`[]Requirement`)?

I want to keep our API objects dumb structs. What's the consequence of changing this to LabelSelector?